src/app/content/utils/seoUtils.ts

```diff
@@ -9,20 +12,182 @@ import {
+    '[data-type="abstract"], .learning-objectives, .chapter-objectives, .be-prepared, .os-teacher'
+    ) || [];
+
+  for (let i = 0; i <= introContentList.length; i++) {
+    if (introContentList[i]) {
+      const parent = introContentList[i].parentNode;
+      if (parent === null) {
+        break;
+      }
+      parent.removeChild(introContentList[i]);
+    }
+  }
+};
+
+export const generateExcerpt = (str: string) => {
+  return str.replace(/\n/g, ' ').replace(/\s\s/g, ' ').trim().substring(0, 152) + '...';
+};
+
+// tslint:disable-next-line: max-line-length
+const getPageType = (node: HTMLElement, values: DescriptionTemplateValues): PageTypes => {
+  const {parentType, chapterTitle, pageTitle} = values;
+  const nodeClasses = node.classList;
+  const nodeType = node.getAttribute('data-type');
+
+  if (nodeType === 'page') {
+    return 'page';
+  } else if (nodeClasses.contains('appendix')) {
+    return 'appendix';
+  } else if (
+    nodeClasses.contains('os-solution-container')
+      || nodeClasses.contains('os-solutions-container')
+      ) {
+    return 'answer-key';
+  } else if (parentType !== 'chapter' && parentType !== 'book') {
+    return 'eoc-sub-page';
+  } else if (pageTitle !== chapterTitle) {
+    return 'eoc-page';
+  } else {
+    return 'eob-page';
+  }
+};
+
+const getPageDescriptionFromContent = (node: HTMLElement): string | null => {
+  if (!node) {
+    return null;
+  }
+  removeIntroContent(node);
+  const firstSection = node.querySelector('section');
+  let para: Element | null = firstSection
+    ? firstSection.querySelector('p')
+    : node.querySelector('p');
+
+  if (!para || !para.textContent) {
+    return null;
+  }
+
+  // Find first <p> of 90+ chars.
+  if (para.textContent.length >= 90) {
+    const mathless = hideMath(para);
+    return mathless ? generateExcerpt(mathless) : null;
+  } else {
+    const text = para.textContent;
+    const next = para.nextElementSibling;
+    while (text && text.length < 90 && next) {
+      if (next.matches('p')) {
+        para = next;
+      }
+    }
+    return null;
+  }
+};
+
+// tslint:disable: max-line-length
+const generateDescriptionFromTemplate = (pageType: PageTypes, values: DescriptionTemplateValues) => {
+  const {parentTitle, pageTitle, bookTitle, chapterTitle} = values;
+  switch (pageType) {
+    case 'page':
+      return `On this page you will discover the ${pageTitle} for ${chapterTitle} of OpenStax's ${bookTitle} free textbook.`;
+    case 'answer-key':
+      return `On this page you will discover the Answer Key for ${pageTitle} of OpenStax's ${bookTitle} free textbook.`;
+    case 'eoc-sub-page':
+      return `On this page you will discover the ${parentTitle}: ${pageTitle} for ${chapterTitle} of OpenStax's ${bookTitle} free textbook.`;
+    case 'eoc-page':
+      return `On this page you will discover the ${pageTitle} for ${chapterTitle} of OpenStax's ${bookTitle} free textbook.`;
+    case 'eob-page' || 'appendix':
```

Comment:
is there a reason to not just generally treat appendix as `eob-page`? as in return `eob-page` from its page type logic?